server/services/auto_tag.py

```
@@ -5,15 +5,61 @@
 
 from __future__ import annotations
 
 import asyncio
 import logging
+import os
+import threading
 import time
+from concurrent.futures import ThreadPoolExecutor
 from datetime import datetime, timezone
-from typing import Any
+from typing import Any, Callable
```

Ruff (0.15.12)

[warning] Import from `collections.abc` instead: `Callable`

Import from `collections.abc`

(UP035)